test/mpi/test_mpi.h

```diff
@@ -94,7 +106,7 @@ static inline const char* skip_str(test_skip_cause_t s) {
     switch(s) {
     case TEST_SKIP_MEM_LIMIT:
         return "maximum buffer size reached";
-    case TEST_SKIP_NOT_IMPL_INPLACE:
+    case TEST_SKIP_NOT_SUPPORTED:
```

Comment:
I have general question. Should we differentiate UCC_ERR_NOT_SUPPROTED and UCC_ERR_NOT_IMPLEMENTED? For example custom reductions in NCCL TL are not supported and likely will not be supported in future that's why return code might be UCC_ERR_NOT_SUPPORTED, in contrast in UCP TL custom reduction will be supported but so far not implemented thus return code is UCC_ERR_NOT_IMPLEMENTED.

Reply:
I agree this is a difference. But from the test perspective: if ucc returns either NOT SUPPORTED or NOT_IMPLEMENTED it is not a test failure imho - there is nothing to fix. This is the expected flow that user has to process properly, right? On the other hand i remember @manjugv was saying that at some point UCC must SUPPORT all the spec. This makes sense as well... Meaning - even if some TL can't support we must still fallback to other path (cl_basic->tl_ucp) which supports everything. But this won't happen soon. I think so far we can do the test that way: lets check both NOT_SUPPORTED and NOT_IMPLEMENTED and consider them a skip reason.

Reply:
@karasevb pls fix error message accordingly

Reply:
Fixed with force-pushed commit